lib/appium_lib/driver.rb

```diff
@@ -376,6 +378,9 @@ def initialize(opts = {})
       @sauce_username   = nil if !@sauce_username || (@sauce_username.is_a?(String) && @sauce_username.empty?)
       @sauce_access_key = appium_lib_opts.fetch :sauce_access_key, ENV['SAUCE_ACCESS_KEY']
       @sauce_access_key = nil if !@sauce_access_key || (@sauce_access_key.is_a?(String) && @sauce_access_key.empty?)
+      @sauce_endpoint   = appium_lib_opts.fetch :sauce_endpoint, ENV['SAUCE_ENDPOINT']
+      @sauce_endpoint   = 'ondemand.saucelabs.com:443/wd/hub' if 
```

Comment:
should be fixed now; i'm confused since my editor is set to remove trailing whitespace, so not sure how this got in here!